go/chunks/chunk_store.go

```diff
@@ -67,6 +67,10 @@ type ChunkStore interface {
 type Factory interface {
 	CreateStore(ns string) ChunkStore
 
+	// CreateStoreFromCache allows caller to signal to the factory that it's
+	// willing to tolerate an out-of-date ChunkStore.
+	CreateStoreFromCache(ns string) ChunkStore
```

Comment:
I think this makes sense for now. It's kind of interesting to think about this contract. The caller of this always has the ability to force a read from "head" via Rebase. You could imagine there being one call which returns you a chunk store and if you care about having reflect the underlying truth as of now, you need to call Rebase() on it yourself.